src/odemis/driver/test/xt_client_test.py

```diff
@@ -281,6 +283,162 @@ def test_set_beam_shift(self):
+                time.sleep(0.2)  # Wait until the auto contrast brightness is finished
+                if time.time() - starting_time > max_execution_time:
+                    break  # If auto contrast brightness takes too long stop waiting
+
+            # Check if the time to perform the auto contrast brightness is not too long
+            self.assertLess(time.time() - starting_time, max_execution_time,
+                            "Execution of auto contrast brightness was stopped because it took more than %s seconds."
+                            % max_execution_time)
+
+            auto_contrast_brightness_state = self.microscope.is_running_auto_contrast_brightness(detector2ChannelName[detector])
+            self.assertEqual(auto_contrast_brightness_state, False)
+
+        # Test if an error is raised when an invalid detector is provided
+        with self.assertRaises(KeyError):
+            self.scanner.applyAutoContrastBrightness("error_expected")
+        time.sleep(2.5)  # Give microscope/simulator the time to update the state
+        autofocus_state = self.microscope.is_autofocusing(detector2ChannelName[detector])
+        self.assertEqual(autofocus_state, False)
+
+    def test_apply_autofocus(self):
+        """
+        Test for the auto functionality of the autofocus.
+        """
+        # Check all the different detector types by looping over them
+        for detector in detector2ChannelName:
+            # Start auto focus and check if it is running.
+            autofocus_future = self.efocus.applyAutofocus(detector)
+            time.sleep(2.5)  # Give microscope/simulator the time to update the state
+            autofocus_state = self.microscope.is_autofocusing(detector2ChannelName[detector])
+            self.assertEqual(autofocus_state, True)
+            self.assertIsInstance(autofocus_future, ProgressiveFuture)
+
+            # Stop auto focus and check if it stopped running.
+            autofocus_future.cancel()
+            time.sleep(5.0)  # Give microscope/simulator the time to update the state
+            autofocus_state = self.microscope.is_autofocusing(detector2ChannelName[detector])
+            self.assertEqual(autofocus_state, False)
+            self.assertIsInstance(autofocus_future, ProgressiveFuture)
+
+            # Start autofocus
+            max_execution_time = 40    # Approximately 3 times the normal expected execution time
+            starting_time = time.time()
+            autofocus_future = self.efocus.applyAutofocus(detector)
+            time.sleep(0.5)  # Give microscope/simulator the time to update the state
+            while self.microscope.is_autofocusing(detector2ChannelName[detector]):
+                time.sleep(0.2)  # Wait until the autofocus is finished
+                if time.time() - starting_time > max_execution_time:
+                    break  # If autofocussing takes too long stop waiting
```

Comment:
Please only use the standard API. I think you can do the same with such code:
``` python
autofocus_future.result(timeout=max_execution_time)
```
This will raise an error on Timeout or if another issue happened.